packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/intrinsics/js/JsPromiseImpl.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024 Elide Technologies, Inc.
+ *
+ * Licensed under the MIT license (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   https://opensource.org/license/mit/
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under the License.
+ */
+@file:Suppress("TooGenericExceptionCaught")
+
+package elide.runtime.gvm.internals.intrinsics.js
+
+import org.graalvm.polyglot.Value
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.Future
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+import java.util.function.Supplier
+import elide.runtime.intrinsics.js.JsPromise
+import elide.vm.annotations.Polyglot
+
+/**
+ * TBD.
+ */
+internal class JsPromiseImpl<T> private constructor (
```

Comment:
Note to self: Needs doc. This is our new implementation of JS/JVM promise interop